x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/ml_inference_logic.ts

```diff
@@ -62,15 +72,18 @@ interface MLInferenceProcessorsActions {
 export interface AddInferencePipelineModal {
   configuration: InferencePipelineConfiguration;
   indexName: string;
+  step: AddInferencePipelineSteps;
```

Comment:
Question (non-blocking): I am a bit confused which properties goes to the where. So is `AddInferencePipelineModal` supposed to have the form information only or else?

Reply:
it’s the UI state of the modal. There is so much other data on the logic I made this to try and simplify just the modal state information needed for create.

There could definitely be an argument made that it’s unnecessary and the data on this could live directly on the logic.